src/blocking.rs

```diff
@@ -70,16 +71,22 @@ where
     ///
     /// Returns an error if the handshake does not proceed. If an error occurs, the connection
     /// instance must be recreated.
-    pub fn open<'v, RNG, Verifier>(
+    pub fn open<'v, Provider>(
         &mut self,
-        context: TlsContext<'v, CipherSuite, RNG>,
+        mut context: TlsContext<'v, Provider>,
     ) -> Result<(), TlsError>
     where
-        RNG: CryptoRng + RngCore,
-        Verifier: TlsVerifier<'v, CipherSuite>,
+        Provider: CryptoProvider<CipherSuite = CipherSuite>,
+        SignatureSize<Provider::SignatureCurve>:
+            core::ops::Add<ecdsa::der::MaxOverhead> + ArrayLength<u8>,
+        ecdsa::der::MaxSize<Provider::SignatureCurve>: ArrayLength<u8>,
```

Comment:
this bound seems to limit to ecdsa or is it implementable by something like RSA as well?

Reply:
Good point. Looking at the signature it looks very `ecdsa` specific :grinning: 

I'll see if I can wrap it into something that will work broader somehow. Any suggestions? 
Most of it is essentially coming from the rustls-rustcrypto crate